packages/common-server/src/git.ts

```diff
@@ -388,12 +388,12 @@ export class GitUtils {
       if (err?.code !== "ENOENT") throw err;
     }
     // Avoid duplicating the gitignore line if it was already there
```

Comment:
noticed dotfile and dotdirs were getting duped because the dot was not in a characer class and not escaped. I briefly thought about fixing this to support all gitignore expressions but that is probably out of scope for this PR 😄 

went the simple way of splitting them by line and checking if it exists in the list.

this PR is still in draft but just wanted to ask if this looks okay or if you would have done it differently @SeriousBug

Reply:
I'd actually do this:

```ts
contents.match(new RegExp(`^${_.escapeRegExp(addPath)}/?$`, "m"))
```

The differences are the escape and the "m" flag (accidentally must have used the "g" flag which is not what we want here)

Reply:
ah didn't think to check if lodash has it already 😅